### PR TITLE
worker/caasoperator: tidy up, add tests

### PR DESCRIPTION
**Pull request by axw**
Removed some bits we can infer, and other things that aren't currently required, and may not be later. Pass Clock in via manifold inputs.

### CODE REVIEW

File: featuretests/agent_caasoperator_test.go [hunk 1]
```diff
@@ -59,19 +60,19 @@ func (s *CAASOperatorSuite) newBufferedLogWriter() *logsender.BufferedLogWriter
...
+func waitForApplicationActive(c *gc.C, dataDir, applicationName string) {
 	timeout := time.After(coretesting.LongWait)
-
 	for {
 		select {
 		case <-timeout:
 			c.Fatalf("no activity detected")
 		case <-time.After(coretesting.ShortWait):
-			link := filepath.Join(agentDir, commands.CommandNames()[0])
+			agentBinaryDir := agenttools.ToolsDir(dataDir, "application-"+applicationName)
```
Comment (wallyworld) [inline featuretests/agent_caasoperator_test.go]: The jujud operator binary doesn't live in a directory named after the application - it is baked into the operator docker image in /usr/bin or somewhere. That's why manifold config had an AgentDir config attribute and was set to os.Args[0]

Comment (axw) [reply featuretests/agent_caasoperator_test.go]: I see. Updated to look in /var/lib/juju/tools, and added a comment on Config.DataDir.

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$

**Pull request merged**